src/collection.ts

```diff
@@ -240,6 +253,51 @@ export class Collection<T extends TigrisCollectionType> {
 		return new Cursor<T>(initializer, this.config);
 	}
 
+	/**
+	 * Performs a query to find a single document in collection. Returns the document if found, else
+	 * null.
+	 *
+	 * @param filter - Query to match the document
+	 * @param readFields - Optional field projection param allows returning only specific document fields in result
+	 * @param tx - Optional session information for transaction context
+	 * @param options - Optional settings for the find query
+	 */
+	findOne(
+		filter: Filter<T>,
+		readFields?: ReadFields,
+		tx?: Session,
+		options?: ReadRequestOptions
+	): Promise<T | undefined> {
+		return new Promise<T>((resolve, reject) => {
+			if (options === undefined) {
+				options = new ReadRequestOptions(1);
+			} else {
+				options.limit = 1;
+			}
+
+			const cursor = this.findMany(filter, readFields, tx, options);
+			const iteratorResult = cursor[Symbol.asyncIterator]().next();
+			if (iteratorResult !== undefined) {
+				iteratorResult
+					.then(
+						(r) => resolve(r.value),
+						(error) => reject(error)
+					)
+					.catch(reject);
+			} else {
+				/* eslint unicorn/no-useless-undefined: ["error", {"checkArguments": false}]*/
+				resolve(undefined);
+			}
+		});
+	}
+
+	/**
+	 * Search for documents in a collection. Easily perform sophisticated queries and refine
+	 * results using filters with advanced features like faceting and ordering.
+	 *
+	 * @param request - Search query to execute
+	 * @param options - Optional settings for search
+	 */
```

Comment:
could you please add such docs on tigris (client) and db class too?

Reply:
Sure! Will cover in next iteration. Will first update starter app and docs to match these changes.